src/handlers/ask-gpt.ts

```diff
@@ -0,0 +1,239 @@
+  const currentIssueKey = createKey(currentIssueUrl, currentIssueNumber);
+  // collect specifically all of the spec and PR bodies
+  const specAndBodies: Record<string, string> = {};
+  specAndBodies[currentIssueKey] = issueSpecOrPullBody || "";
+  specAndBodies[specReferencedIssueKey as string] = specReferencedIssueBody || "";
+
+  for (const linkedIssue of linkedIssues) {
+    const issue = await fetchIssue({ context, issueNum: linkedIssue.issueNumber, repo: linkedIssue.repo });
+    const { body, repository_url, pull_request } = issue;
+    const linkedIssueKey = createKey(repository_url, linkedIssue.issueNumber);
+    specAndBodies[linkedIssueKey] = body || "";
+
+    if (pull_request) {
+      linkedPulls[linkedIssueKey] = true;
+    }
+  }
+
+  return { specAndBodies, linkedPulls };
+}
+
+async function getSpecReferencedContext(context: Context, org: string, issueSpecOrPullBody: string | null) {
+  // fetch the spec referenced issue if it exists
+  const specReferencedIssueId = idIssueFromComment(org, issueSpecOrPullBody);
+  let specReferencedIssue,
+    specReferencedIssueBody,
+    specReferencedIssueRepoUrl,
+    specReferencedIssueComments,
+    specReferencedIssueKey,
+    streamlinedSpecReferencedIssueComments;
+
+  if (specReferencedIssueId) {
+    specReferencedIssue = await fetchIssue({ context, issueNum: specReferencedIssueId.issueNumber });
+    specReferencedIssueBody = specReferencedIssue.body;
+    specReferencedIssueRepoUrl = specReferencedIssue.repository_url;
+    specReferencedIssueComments = await fetchIssueComments({ context, issueNum: specReferencedIssueId.issueNumber, repo: specReferencedIssueId.repo });
+    specReferencedIssueKey = createKey(specReferencedIssueRepoUrl, specReferencedIssueId?.issueNumber);
+    streamlinedSpecReferencedIssueComments = streamlineComments(specReferencedIssueComments) ?? {};
+  }
+
+  return {
+    specReferencedIssue,
+    specReferencedIssueBody,
+    specReferencedIssueRepoUrl,
+    specReferencedIssueComments,
+    specReferencedIssueKey,
+    streamlinedSpecReferencedIssueComments,
+  };
+}
+
+function createKey(issueUrl: string, issue?: number) {
+  const splitUrl = issueUrl?.split("/");
+  const issueNumber = issue || parseInt(splitUrl?.pop() || "");
+  const issueRepo = splitUrl?.slice(-2).join("/");
+  const issueOrg = splitUrl?.slice(-3, -2).join("/");
+
+  if (issueOrg.startsWith("repos")) {
+    return `${issueRepo}/issues/${issueNumber}`;
+  }
+
+  return `${issueOrg}/${issueRepo}/${issueNumber}`;
+}
+
+function streamlineComments(comments: IssueComments) {
+  const streamlined: Record<string, StreamlinedComment[]> = {};
+
+  for (const comment of comments) {
+    const user = comment.user?.login;
+    if (
+      user === "ubiquibot" ||
+      user === "ubiquibot[bot]" ||
+      user === "ubiquibot-v2-testing" ||
+      user === "ubiquibot-dev[bot]" ||
+      user === "ubqbot[bot]" || // TODO: remove this
+      user === "github-actions[bot]"
+    ) {
```

Comment:
Definitely just ignore all bot comments as we do in conversation rewards.

Reply:
> Should this plugin be able to read it's own comments or not?

Without it seeing it's own answers it may try to re-answer a previous question and would be without relevant context. Originally I had it able to read it's own comments and had planned to do the same here

Reply:
I prefer to exclude for now.